.github/workflows/cdk-e2e.yaml

```diff
@@ -36,13 +36,13 @@ jobs:
           fi
 
       - name: Checkout AWS CDK main branch
-        uses: actions/checkout@v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
These hashes will be auto-updated by dependabot, including the comment. So we shouldn't have to worry manual updates. The intention is to track exactly one version of the actions (linked to an unchangable hash), rather than a generic version like `@v4`, which could potentially point to a malicious commit without us knowing.